packages/eslint-plugin/src/rules/naming-convention-utils/parse-options.ts

```diff
@@ -87,6 +87,7 @@ function parseOptions(context: Context): ParsedOptions {
   const result = util.getEnumNames(Selectors).reduce((acc, k) => {
     acc[k] = createValidator(k, context, normalizedOptions);
     return acc;
+    // eslint-disable-next-line @typescript-eslint/prefer-reduce-type-parameter
```

Comment:
i wonder if this should be an exception in the rule?
I don't think `{}` can ever be typed with the parameter, can it?

Reply:
https://github.com/typescript-eslint/typescript-eslint/issues/3440#issuecomment-1356641311 

... https://github.com/microsoft/TypeScript/issues/9879 🙃